src/ServiceInsight/Framework/WorkNotifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace ServiceInsight.Framework
+{
+    using System;
+    using System.Reactive.Disposables;
+    using Caliburn.Micro;
+    using ServiceInsight.Framework.Events;
+
+    public interface IWorkNotifier
+    {
+        IDisposable NotifyOfWork();
+
+        IDisposable NotifyOfWork(string startMessage);
+
+        IDisposable NotifyOfWork(string startMessage, string finishMessage);
+    }
+
+    class WorkNotifier : IWorkNotifier
+    {
+        IEventAggregator eventAggregator;
+
```

Comment:
@HEskandari Firstly, the shell isn't the only thing that implements `IWorkTracker`.

I see your point, why even use the EventAggregator as a middle-man? Why not have ViewModels that care about work status "subscribe" directly to the WorkNotifier? We could, but that would be a bigger change. I'll keep that idea in mind though.